irods/manager/data_object_manager.py

```diff
@@ -126,21 +130,23 @@ def put(self, local_path, irods_path, return_data_object = False, num_threads =
         else:
             obj = irods_path
 
-        with open(local_path, 'rb') as f, self.open(obj, 'w', **options) as o:
-
-            if self.should_parallelize_transfer (num_threads, f):
+        with open(local_path, 'rb') as f:
+            sizelist = []
+            if self.should_parallelize_transfer (num_threads, f, measured_obj_size = sizelist):
```

Comment:
`f` is passed as the `obj_sz`?

reading a bit more of the code and alan's comment, i'm guessing `obj_sz` is treated as a variant-like type?

Reply:
Yes, it's passed as the obj_sz.  (This function is written so that a file or integer-like entity can be placed in that parameter - I forget exactly why though...)